cpp/src/io/utilities/config_utils.cpp

```diff
@@ -244,23 +244,22 @@ CUDF_EXPORT std::mutex& host_mr_mutex()
 }
 
 // Must be called with the host_mr_mutex mutex held
-CUDF_EXPORT rmm::host_async_resource_ref& make_host_mr(std::optional<host_mr_options> const& opts)
+CUDF_EXPORT std::pair<rmm::host_async_resource_ref*, bool> make_host_mr(
```

Comment:
What? Pointer to reference?

Reply:
Okay, this seems legitimate. BTW, does this need to be thread-safe?

Reply:
oh, missed this. Could this not have stayed as a reference?

Reply:
> BTW, does this need to be thread-safe?

No, it's always called within a critical section (I think that's the term)

Reply:
In order to return a reference in `std::pair`, it may need to be wrapped into a reference wrapper, which looks ugly:
```
std::pair<std::reference_wrapper<rmm::host_async_resource_ref>, bool> ...
```

Reply:
So if copying is not expensive for `rmm::host_async_resource_ref` (which I assume so, because this class is passed by copy everywhere), I'm fine with just copying it.

Reply:
there was a specific reason I used a reference initially
need to look at this again to remember

Reply:
we keep a static pointer to host_async_resource_ref in make_host_mr
so we can't return a copy from make_default_pinned_mr, since the copy will go out of scope.
That said, looks like make_host_mr can return a copy

Reply:
I'll go back to the bool* approach we had before, especially since this is all internally to this file.

Reply:
Reverted it, please see latest.